app/containers/ItunesContainer/reducer.ts

```diff
@@ -0,0 +1,46 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { song } from './types';
+
+export interface ituneResponseState {
+  loading: boolean;
+  songData: {
+    resultCount: number;
+    results: song[];
+  };
+  songListError: string;
+  searchTerm: string;
+}
+
+export const initialState: ituneResponseState = {
+  loading: false,
+  songData: { resultCount: 0, results: [] },
+  songListError: '',
+  searchTerm: ''
```

Comment:
Please remove this property if it's unused

Reply:
I am using searchTerm state to store keyword and passing this as param in api call

Reply:
can't we use the action payload ?